### DISCUSSION TITLE
sandbox应该如何使用呢

### DISCUSSION THREAD

**Discussion by superchenwb** [alibaba/designable#22] [Q&A]
`<Sandbox
    jsAssets={['./runtime.bundle.js', './sandbox.bundle.js']}
  />`
Sandbox组件现在可以使用吗？
runtime.bundle.js和sandbox.bundle.js这两个js是怎么来的呢

**Comment by janryWang** [comment]
把 https://github.com/alibaba/designable/blob/main/packages/playground/src/sandbox.tsx 单独打成一个bundle就行了

**Reply by superchenwb** [reply]
这个我试过了，但是runtime.bundle.js还是不知道从哪来的？
而且在拖拽组件到面板后，并没有显示出组件来；
另外就是usePrefix无法取到prefixCls的值：
![image](https://user-images.githubusercontent.com/17635660/126602612-3fb25861-bafd-4219-a023-5efd11384071.png)

**Reply by superchenwb** [reply]
这是我修改的打包文件
![image](https://user-images.githubusercontent.com/17635660/126602811-d5413c71-6f9a-4b94-b45b-6ef5993320d1.png)

**Reply by janryWang** [reply]
不需要runtime，这跟webpack打包相关，之前我是做了split

**Reply by superchenwb** [reply]
那拖拽组件到中间面板时，iframe并没有更新，所以看起来是空的，既然用到了iframe，那应该是需要通讯的吧，不知道designable是如果做通讯的

**Reply by janryWang** [reply]
有没有报错

**Reply by superchenwb** [reply]
没有报错，但是tree是有数据的，但是我不知道iframe中是否存在数据，还是说iframe页面没有更新

**Reply by superchenwb** [reply]
![image](https://user-images.githubusercontent.com/17635660/126617015-9d25bb8f-1c2e-4455-8090-17ed86baa690.png)

**Reply by janryWang** [reply]
哦，reactive需要exteranls掉，统一在html层加载umd

**Reply by janryWang** [reply]
因为我跨iframe通讯是通过共享reacitve达到的

**Reply by superchenwb** [reply]
额，不明白是如何通过共享reacitive达到共享的，能具体讲一下吗，或者类似的文章可以让我看一下

**Reply by superchenwb** [reply]
将reactive添加到externals后，统一在html加载umd，确实可以了，但是其中的逻辑还是没有想通，我再好好思考一下

**Reply by superchenwb** [reply]
还有一个问题，就是css的问题，需要把整个playground.css加载到iframe中吗

**Reply by janryWang** [reply]
不需要，sandbox里面用到哪些组件，就依赖哪些组件的样式就行

**Reply by superchenwb** [reply]
在useDesigner中获取到Engine对象实例，在top和iframe中拿到的实例其实是同一个对吗，当observable对象发生变化时，会同时更新top和iframe中的组件，不知道这样的理解有没有问题

**Reply by janryWang** [reply]
嗯，没毛病

**Reply by superchenwb** [reply]
谢谢啦

**Comment by Evan0307** [comment]
麻烦请教下，<Sandbox jsAssets={['./runtime.bundle.js', './sandbox.bundle.js']} /> 这是在哪里渲染，

<ViewPanel type="DESIGNABLE">
       {() => <Sandbox  jsAssets={['./sandbox.bundle.js']}/>}
 </ViewPanel>
  现在代替原来的Content 渲染 会报错

![image](https://user-images.githubusercontent.com/39261372/137271497-7204d006-3221-4cf2-99b5-a06c870d5ca9.png)
